sdk/orm-django/django_encryption/fields.py

```diff
@@ -42,8 +44,163 @@ class _RaiseError:
+                setattr(instance, 'decrypted_' + self.field.name, None)
+                return None
+
+            transformation = None
+            if hasattr(instance, '_transform_fields'):
+                transformation = instance._transform_fields.get(
+                    self.field.name)
+            decrypted_value = self.field.get_decrypted_value(
+                encrypted_value, transformation=transformation)
+            setattr(instance, 'decrypted_' + self.field.name, decrypted_value)
+            return decrypted_value
+        return DeferredAttribute.__get__(self, instance, owner)
+
+    def __set__(self, instance, value):
+        if instance is None:
+            raise AttributeError("Can only be set on instances")
+        if value is None:
+            setattr(instance, 'decrypted_' + self.field.name, None)
+            setattr(instance, 'encrypted_' + self.field.name, None)
+            return
+        if isinstance(value, tuple):
+            # we got an encrypted value
+            value = value[1]
+            setattr(instance, 'encrypted_' + self.field.name, value)
+            return
+        # we got a decrypted value
+        setattr(instance, 'decrypted_' + self.field.name, value)
+
+    # get_prefetch_queryset is called by django when prefetching related objects
+    # this function allows django's queries to believe that the encrypted fields
+    # are like foreign keys and so can be prefetched
+    def get_prefetch_queryset(self, instances, queryset=None):
+        encrypted_attr_name = 'encrypted_' + self.field.name
+        encrypted_values = [getattr(instance, encrypted_attr_name)
+                            for instance in instances]
+        transformation = None
+        # we could actually have a separate transformation for each instance, but we are assuming
+        # that it's the same transformation for all of them
+        if hasattr(instances[0], '_transform_fields') and self.field.name in instances[0]._transform_fields:
+            transformation = instances[0]._transform_fields.get(
+                self.field.name)
+        rel_qs = self.field.get_decrypted_values(
+            encrypted_values, transformation=transformation)
+
+        def rel_obj_attr(obj):
+            return 'decrypted_' + self.field.name
+
+        def instance_attr(obj):
+            return 'decrypted_' + self.field.name
+
+        single = True
+        cache_name = 'decrypted_' + self.field.name
+        is_descriptor = True
+        return (
+            rel_qs,
+            rel_obj_attr,
+            instance_attr,
+            single,
+            cache_name,
+            is_descriptor)
+
+    def is_cached(self, obj):
+        result = hasattr(obj, 'decrypted_' + self.field.name) and getattr(obj,
+                                                                          'decrypted_' + self.field.name) is not None
+        return result
+
 
 class EncryptedMixin(object):
+    is_relation = True
```

Comment:
I don't see where you are using is_relation

Reply:
I thought it was needed to convince django my field was a foreign key, your comment made me check that again, and remove it, as it's unnecessary. Thanks!